tensorboard/plugins/example/tensorboard_plugin_example/plugin.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""A sample plugin to demonstrate dynamic loading."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+
+import numpy as np
+import six
+from tensorboard.plugins import base_plugin
+from tensorboard.util import tensor_util
+import werkzeug
+from werkzeug import wrappers
+
+from tensorboard_plugin_example import metadata
+
+
+class ExamplePlugin(base_plugin.TBPlugin):
+  plugin_name = metadata.PLUGIN_NAME
+
+  def __init__(self, context):
+    self._multiplexer = context.multiplexer
+
+  def is_active(self):
+    return bool(self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME))
+
+  def get_plugin_apps(self):
+    return {
+        "/index.js": self._serve_js,
+        "/tags": self._serve_tags,
+        "/greetings": self._serve_greetings,
+    }
+
+  def frontend_metadata(self):
+    return super(ExamplePlugin, self).frontend_metadata()._replace(
+        es_module_path="/index.js",
+    )
+
+  def _tags_impl(self):
```

Comment:
For an example, is there any particular reason not to just inline these into the request handler methods?  They're not particularly long and it's not like we're testing them separately.  I could see would-be developers scratching their heads and not understanding the point of factoring this out into a separate method.

Reply:
Sure; can do.